design/client-secret-rotation.md

```diff
@@ -78,3 +83,13 @@ The changes needed in the Admin UI for a client would be:
 This approach allows a global control over client secrets within a realm, and has less changes to the APIs, and is fully
 backwards compatible as there would be no change to the default behaviour. It also fits better into dynamic client
 registration than the previous proposal.
+
+Admin notification events
+-------------------------
+
+Throw an administrative events related to rotation and expiration of secrets.
+- When a client rotates a secret. Client ID and Client Name / Description.
+- When trying to authenticate using an expired rotated secret.
```

Comment:
Currently when authentication with invalid client secret is done, the "normal" event is triggered, but not the "admin event" . AFAIK it is the normal error event with error like "invalid client authentication" . I guess this should not be different when authentication with expired rotated secret is done?

Maybe we can add some "detail" to the event that invalid client authentication was tried with expired "rotated" secret. But IMO it should be same event like during failed client authentication with expired "regular" client secret.

Reply:
Yes, it is a normal event. I was wrong when I mentioned administrative event. My suggestion is to add customer id and their description/name as detail.

Reply:
Hmm... I am not sure what do you mean by "customer id" ? Didn't you mean "client id" ? :-)

Reply:
Yeah!! sorry again. That's what I meant, not customer, but **client**